### PR TITLE
start modularizing slick.grid.js

### PR DESCRIPTION
**Pull request by ghost**
Aside from changes in `README.md`, `package.json` & `bower.json`, these are the steps I've taken:
1. Move `slick.grid.js` to `src/grid/index.js`.
2. Remove `this.eval` from `index.js` (rollup.js doesn't like it, plus it's a potential vector).
3. Remove top-level IIFE and unindent the code (since rollup does that for you).
4. Added `var $ = jQuery;` to replace what IIFE did.

`slick.grid.js` is the generated output of `npm run build`.

I know this PR doesn't do any actual modularization.  I started moving the hidden columns code, but soon realized it wasn't a 5min change, so I'll do it later (i.e. WIP at https://github.com/coatue/SlickGrid/commit/e19da544f6eea96a7bdedfd8ebcb794da887b353).

Next step would be to get travis going, where it'll run the build to determine whether the latest bundle was committed or not (and pass/fail accordingly).

### CODE REVIEW

**Comment by bcherny** [comment]
want to use circleci instead of travis? :)

File: package.json [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
-  "version": "3.5.0",
+  "version": "3.5.1",
+  "scripts": {
+    "build": "rollup -f iife -o slick.grid.js src/grid/index.js"
+  },
   "devDependencies": {
     "browser-sync": "^1.6.0",
     "coffee-script": "^1.8.0",
-    "gulp": "^3.8.9"
+    "gulp": "^3.8.9",
```
Comment (bcherny) [inline package.json]: is this used?

**Comment by bcherny** [comment]
:+1:

File: package.json (hunk 1 above)
Comment (ghost) [inline package.json]: Is it?  If not, I can remove it (plus `gulpfile.coffee/js`).

Comment (bcherny) [inline package.json]: doesnt seem to be!

**Comment by ghost** [comment]
I only recommended travis since that's what I'm familiar with.  Happy to go with circleci - is it free for open source repos?

File: package.json (hunk 1 above)
Comment (ghost) [inline package.json]: OK (I never liked it anyway), I'll 'nix it and if anyone complains, it's easy to put back.

**Comment by ghost** [comment]
OK, cleaned gulp, etc. up in the latest SHA: https://github.com/coatue/SlickGrid/commit/0c539afb42bcf2eeb464d0be270ba67e380b1ca3

**Pull request merged**